src/test/groovy/org/prebid/server/functional/tests/privacy/AuctionGppSpec.groovy

```diff
@@ -0,0 +1,226 @@
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.regs.gdpr == 0
+        assert bidderRequest.regs.gppSid == gppSidsId
+    }
+
+    def "PBS should emit warnings when regs.gpp_sid contains 2 and regs.gdpr isn't 1"() {
+        given: "Default bid request with gppSid and gdpr"
+        def gdpr = 0
+        def gppSectionsId = [TCF_EU_V2.valueAsInt]
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs.gdpr = gdpr
+            regs.gppSid = gppSectionsId
+        }
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain warning"
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.code } == [999]
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.message } ==
+                ["GPP scope does not match TCF2 scope"]
+
+        and: "Bidder request should contain the same value as in request"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.regs.gdpr == gdpr
+        assert bidderRequest.regs.gppSid == gppSectionsId
+    }
+
+    def "PBS should emit warnings when regs.gpp_sid not contains 2 and regs.gdpr isn't 0"() {
+        given: "Default bid request with gppSid and gdpr"
+        def gdpr = 1
+        def gppSidsId = [PBSUtils.getRandomNumberWithExcept(TCF_EU_V2.valueAsInt)]
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs.gdpr = gdpr
+            regs.gppSid = gppSidsId
+        }
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain warning"
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.code } == [999]
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.message } ==
+                ["GPP scope does not match TCF2 scope"]
+
+        and: "Bidder request should contain the same value as in request"
+        def resolvedRequest = response.ext.debug.resolvedRequest
+        assert resolvedRequest.regs.gdpr == gdpr
+        assert resolvedRequest.regs.gppSid == gppSidsId
+    }
+
+    def "PBS should emit warning when GPP string is invalid"() {
+        given: "Default bid request with invalid gpp"
+        def invalidGpp = "Invalid_GPP_Consent_String"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs = new Regs(gpp: invalidGpp)
+        }
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain warning"
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.code } == [999]
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.message } ==
+                ["GPP string invalid: Undecodable FibonacciIntegerRange '101111'"]
```

Comment:
I wouldn't code this. This text will most likely change as GPP library matures. Match only to the `GPP string invalid: *`